### PR TITLE
Expose GetFinetuneMetrics endpoint for finetunes

### PR DESCRIPTION
**Pull request by innainu**
I think the user would want to see the progression of metrics during training, so kept the whole json to display.

```
In [5]: co.get_model_metrics('624b36b4-3ba5-4a0b-84b5-881f1adf6e62')
Out[5]: 
[ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 52, 34, 546308, tzinfo=datetime.timezone.utc), step_num=0, loss=4.8585205, accuracy=0.26640624),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 55, 37, 43164, tzinfo=datetime.timezone.utc), step_num=16, loss=3.408817, accuracy=0.48359376),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 56, 4, 762593, tzinfo=datetime.timezone.utc), step_num=32, loss=2.4887414, accuracy=0.5921875),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 56, 32, 459780, tzinfo=datetime.timezone.utc), step_num=48, loss=1.5785851, accuracy=0.62890625),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 57, 0, 98393, tzinfo=datetime.timezone.utc), step_num=64, loss=1.477885, accuracy=0.625),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 57, 27, 798969, tzinfo=datetime.timezone.utc), step_num=80, loss=1.4387118, accuracy=0.6125),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 57, 55, 423252, tzinfo=datetime.timezone.utc), step_num=96, loss=1.4775631, accuracy=0.62109375),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 58, 23, 1025, tzinfo=datetime.timezone.utc), step_num=112, loss=1.6538131, accuracy=0.6273438),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 58, 50, 533064, tzinfo=datetime.timezone.utc), step_num=128, loss=1.9606082, accuracy=0.6125),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 59, 18, 351986, tzinfo=datetime.timezone.utc), step_num=144, loss=2.153335, accuracy=0.609375),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 21, 59, 45, 935886, tzinfo=datetime.timezone.utc), step_num=160, loss=2.4700398, accuracy=0.625),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 22, 0, 14, 285835, tzinfo=datetime.timezone.utc), step_num=176, loss=2.651921, accuracy=0.6421875),
 ModelMetric(created_at=datetime.datetime(2023, 4, 13, 22, 0, 41, 906381, tzinfo=datetime.timezone.utc), step_num=192, loss=2.775621, accuracy=0.63593745)]

```

### CODE REVIEW

File: cohere/responses/custom_model.py [hunk 1]
```diff
@@ -73,6 +74,23 @@ def from_dict(cls, data: Dict[str, Any]) -> "CustomModel":
             model_id=data["model"]["route"] if "model" in data else None,
         )
 
+@dataclass
```
Comment (innainu) [inline cohere/responses/custom_model.py:77]: might be easier to have `CustomModel` as a dataclass as well, would omit having to write out the init method.

File: cohere/responses/custom_model.py [hunk 2]
```diff
@@ -73,6 +74,23 @@ def from_dict(cls, data: Dict[str, Any]) -> "CustomModel":
...
+class ModelMetric(CohereObject):
+    created_at: datetime
+    step_num: int
+    loss: float
+    accuracy: float
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "ModelMetric":
+        # model metrics timestamp sometimes contains nanoseconds, so we truncate
+        dt_concat = data["created_at"][:26] + data["created_at"][-1:]
```
Comment (innainu) [inline cohere/responses/custom_model.py:87]: this is cringy, but for some reason the timestamps back from the metrics.json is variable up to nanoseconds.. I can instead use a different library (I see dateutil.parser can handle variable seconds like this), or if anyone has a different suggestion here.

File: cohere/client.py [hunk 3]
```diff
@@ -1051,6 +1052,18 @@ def get_custom_model_by_name(self, name: str) -> CustomModel:
         response = self._request(f"{cohere.CUSTOM_MODEL_URL}/GetFinetuneByName", method="POST", json=json)
         return CustomModel.from_dict(response["finetune"])
 
+    def get_model_metrics(self, custom_model_id: str) -> List[ModelMetric]:
```
Comment (raphaelcristal) [inline cohere/client.py:1055]: should we call it get_custom_model_metrics? All of the other functions start with custom_model_....

File: cohere/responses/custom_model.py (hunk 2 above)
Comment (raphaelcristal) [reply cohere/responses/custom_model.py:88]: honestly I have no better suggestion here other than at least pulling this logic into a function with a test, so it is more visible why it was done like this.

Comment (innainu) [reply cohere/responses/custom_model.py:91]: k, I'll add the test. wanted to see if there was disagreement before continuing

**Review by mkozakov — APPROVED** [review]
is `tests/__init__.py` needed?

File: pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "cohere"
-version = "4.17.0"
+version = "4.18.0"
```
Comment (mkozakov) [inline pyproject.toml:3]: i think you need to bump this again

Comment (raphaelcristal) [reply pyproject.toml]: done

**Comment by raphaelcristal** [comment]
> is `tests/__init__.py` needed?

deleted

**Review by mkozakov — APPROVED** [review]

**Pull request merged**